File: module_mediator/src/main/java/main/ClassUtil.java
package main;

import android.app.Activity;
import android.support.v4.app.Fragment;
import android.view.View;

import java.lang.reflect.Constructor;
import java.lang.reflect.InvocationTargetException;

/**
 * 类工具
 */
@SuppressWarnings({"WeakerAccess", "unused"})
public class ClassUtil {

    private ClassUtil() { }

    public static <T> T newInstance(Class<T> clazz) {
        T instance = null;
        Constructor<?>[] cs = clazz.getDeclaredConstructors();
        try {
            //noinspection unchecked
            instance = (T) cs[0].newInstance();
        } catch (InstantiationException | InvocationTargetException | IllegalAccessException e) {
            e.printStackTrace();
        }
        return instance;
    }

    public static boolean isAndroidUiClass(Class targetClass) {
        return Activity.class.isAssignableFrom(targetClass)
                || View.class.isAssignableFrom(targetClass)
                || Fragment.class.isAssignableFrom(targetClass)
                || android.app.Fragment.class.isAssignableFrom(targetClass);
    }
}

File: module_mediator/src/main/java/main/ClassUtil.kt
package main

import android.app.Activity
import android.support.v4.app.Fragment
import android.view.View

import java.lang.reflect.Constructor
import java.lang.reflect.InvocationTargetException

/**
 * 类工具
 */
object ClassUtil {

    @Suppress("UNCHECKED_CAST")
    fun <T> newInstance(clazz: Class<T>): T? {
        var instance: T? = null
        val cs = clazz.declaredConstructors
        try {

            instance = cs[0].newInstance() as T
        } catch (e: InstantiationException) {
            e.printStackTrace()
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }

        return instance
    }

    fun isAndroidUiClass(targetClass: Class<*>): Boolean {
        return (Activity::class.java.isAssignableFrom(targetClass)
                || View::class.java.isAssignableFrom(targetClass)
                || Fragment::class.java.isAssignableFrom(targetClass)
                || android.app.Fragment::class.java.isAssignableFrom(targetClass))
    }
}